packages/blocks/src/api/factory.js

```diff
@@ -344,6 +344,43 @@ export function getPossibleBlockTransformations( blocks ) {
 	] );
 }
 
+/**
+ * Returns a subset of the transforms from `getPossibleBlockTransformations`.
+ * This subset was selected to support transforms that the mobile apps support,
+ * and to avoid "confusing" transforms that don't feel intuitive in a mobile
+ * context.
+ *
+ * @param {Array} block The block to transform.
+ *
+ * @return {Array} Block types that the block argument can be transformed to in
+ * the mobile app.
+ */
+export function getMobileBlockTransformations( block ) {
```

Comment:
Have you considered hooking into block registration on mobile and picking only subset of transforms instead? You could either write a filter that updates the list of registered transforms with:

https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/filters/block-filters.md#blocksregisterblocktype

or you could extract existing transforms to their own files and override them with a native version of the file.

In general, having a mobile specific name of the same functionality is confusing here.

Reply:
What @gziolo mentioned here is more of what I had in mind from our side conversation @blixt-io.